packages/vm/src/evm/opcodes/codes.ts

```diff
@@ -322,5 +325,36 @@ export function getOpcodesForHF(common: Common): OpcodeList {
     opcodeBuilder[key].fee = common.param('gasPrices', opcodeBuilder[key].name.toLowerCase())
   }
 
+  if (customOpcodes) {
+    for (const _code of customOpcodes) {
+      const code = <any>_code
+      if (code.logicFunction === undefined) {
+        delete opcodeBuilder[code.opcode]
+        continue
+      }
+
+      // Sanity checks
+      if (code.opcodeName === undefined || code.baseFee === undefined) {
```

Comment:
Do we want extra checks here? What if I try to delete an opcode which is not activated, should it throw? What if I try to add opcodes at the same opcode number twice?

Reply:
I guess it provides more freedom if it just deletes an opcode if the opcode is there and otherwise does nothing rather than throwing. Let's imagine the case where poeple just haven't included opcode x in their sidechain at all and in a normal mainnet context the opcode gets activated along a HF. Then this would throw here when used in a generic context and the VM gets set before this HF. I guess that's not what we want and rather generally not necessary to include this limitation.

Hmm, I guess the "opcode twice" case might also not need extra handling. People should just intuitively grasping that they shouldn't really be doing this. And if this happens for whatever reasons then simply the second implementation is taken. For the "opcode twice" case I guess it wouldn't hurt either though to add a simple check if easily doable.